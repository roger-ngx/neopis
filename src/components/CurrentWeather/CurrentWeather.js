import React from 'react';
import './CurrentWeather.css'
import image from '../../assets/images/weather.svg'

const CurrentWeather = props => {
  return <div className='weather'>
    <img src={image} className='weather_icon' alt='weather icon' />
    <div className='weather_temperature'>
      {props.temperature}
    </div>
    <div className='weather_humidity'>
      {props.humidity}
    </div>
  </div>
}

export default CurrentWeather;