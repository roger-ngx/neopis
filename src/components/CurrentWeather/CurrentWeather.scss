.weather {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  min-width: 120px;
}

.weather_icon {
  width: 32px;
  height: 28px;
  object-fit: contain;
}

.weather_temperature {
  width: 30px;
  font-family: HiraKakuProN-W6;
  font-size: 16px;
  letter-spacing: -0.5px;
  text-align: center;
  color: #ffffff;
}

.weather_humidity {
  width: 22px;
  font-family: HiraKakuProN-W3;
  font-size: 14px;
  letter-spacing: -0.4px;
  text-align: center;
  color: #9e9ea5;
}