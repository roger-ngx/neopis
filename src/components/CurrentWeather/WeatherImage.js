import React from 'react';
const images = require.context('../../assets/images/weather');

const WeatherImg = (props) => {
    function getWeatherIconName() {
        function _isDayTime() {
            const hour = (new Date()).getHours();
            return hour > 6 && hour < 18;
        }

        if (!props.weather) {
            return null;
        }

        let imageName = props.weather;

        if (!_isDayTime()) {
            imageName += '_NIGHT';
        }

        imageName += '.png';

        return images(`./${imageName}`);
    };

    return <img src={getWeatherIconName()} className='weather_icon' alt='weather icon' />
}

export default WeatherImg;
