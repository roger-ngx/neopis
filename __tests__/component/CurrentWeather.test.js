import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import CurrentWeather, {CurrentWeather_} from '../../src/components/CurrentWeather/CurrentWeather'

Enzyme.configure({ adapter: new Adapter() })

describe("CurrentWeather", () => {
  it('Render without error', async() => {
    const elem = await renderer
    .create(<CurrentWeather_ temperature={25} humidity={75}/>)
    .toJSON();

    expect(elem).toMatchSnapshot();
  });
});

describe("<CurrentWeather> UI Component", () => {
  const initialState = {weather: {temperature: 0, humidity: 0}};
  const mockStore = configureStore();
  let store, container, dumbContainer;

  beforeAll(() => {
    store = mockStore(initialState);
    container = shallow(<CurrentWeather store={store}/>);
    dumbContainer = shallow(<CurrentWeather_ temperature={25} humidity={75}/>);
  })
  it('render default connect CurrentWeather', () => {
    expect(container.prop('temperature')).toEqual(initialState.weather.temperature);
  });

  it('render default dumb CurrentWeather', () => {
    expect(dumbContainer.find('div.weather_temperature').text()).toEqual('25Â°C');
  });
})