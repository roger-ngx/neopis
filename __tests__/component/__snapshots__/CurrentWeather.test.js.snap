// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather Render without error 1`] = `
<div
  className="weather"
>
  <img
    alt="weather icon"
    className="weather_icon"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <div
    className="weather_temperature"
  >
    25
    Â°C
  </div>
  <div
    className="weather_humidity"
  >
    75
    %
  </div>
</div>
`;
